### PR TITLE
Add tico model

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

* Add TiCo model code for ImageNet benchmark
* Run experiments and add results to readme and benchmarks tab in the docs

Note that I'll have to rebase this PR on master as quite some time passed since I started this.

### CODE REVIEW

**Comment by IgorSusmelj** [comment]
This will be a more tough one. I was running several experiments and the loss is going up and accuracy stays down. Somehow the training seems unstable. I varied some of the hyperparameters but didn't get it running.

**Comment by IgorSusmelj** [comment]
The training on ImageNet does not work as expected. We did several modifications to the loss but no matter what we do it ends up with having the loss saturing to the max value and the accuracy staying at `0%`.
```
Epoch 3:  63%|██████▎   | 1583/2502 [14:05<08:10,  1.87it/s, v_num=0, train_loss=8.000, val_online_cls_loss=140.0, val_online_cls_top1=0.00106, val_online_cls_top5=0.00414]
```

**Comment by guarin** [comment]
@IgorSusmelj could you quickly summarize the changes you tried?

**Comment by IgorSusmelj** [comment]
The latest changes seem promising:
image

File: README.md [hunk 1]
```diff
@@ -256,85 +252,81 @@ trainer.fit(model, dataloader)
...
+| SimCLR\* + DCLW | Res50    | 256        | 100    | 64.5        | 73.2          | 48.5     | [link](https://tensorboard.dev/experiment/TrALnpwFQ4OkZV3uvaX7wQ/) | [link](https://lightly-ssl-checkpoints.s3.amazonaws.com/imagenet_resnet50_dclw_2023-07-07_14-57-13/pretrain/version_0/checkpoints/epoch%3D99-step%3D500400.ckpt)        |
+| SwAV            | Res50    | 256        | 100    | 67.2        | 75.4          | 49.5     | [link](https://tensorboard.dev/experiment/Ipx4Oxl5Qkqm5Sl5kWyKKg)  | [link](https://lightly-ssl-checkpoints.s3.amazonaws.com/imagenet_resnet50_swav_2023-05-25_08-29-14/pretrain/version_0/checkpoints/epoch%3D99-step%3D500400.ckpt)        |
+| TiCo            | Res50    | 256        | 100    | 49.7        | 72.7          | 26.6     | -                                                                  | [link](https://lightly-ssl-checkpoints.s3.amazonaws.com/imagenet_resnet50_tico_2024-01-07_18-40-57/pretrain/version_0/checkpoints/epoch%3D99-step%3D250200.ckpt)        |
```
Comment (guarin) [inline README.md:283]: Interesting that the linear accuracy is so low. In the paper they report linear accuracy that is similar to the other methods.

**Review by guarin — APPROVED** [review]
LGTM! Let's update the benchmarks once we figure out what is wrong with the model.

I'll also create follow-up issues to update BYOL and MoCo to use the correct backbone for evaluation.

File: benchmarks/imagenet/resnet50/tico.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                },
+                {
+                    "name": "online_classifier",
+                    "params": self.online_classifier.parameters(),
+                    "weight_decay": 0.0,
+                },
+            ],
+            # Settings follow original code for 100 epochs which are slightly different
+            # from the paper, see:
+            # https://github.com/deepmind/deepmind-research/blob/f5de0ede8430809180254ee957abf36ed62579ef/byol/configs/byol.py#L21-L23
```
Comment (guarin) [inline benchmarks/imagenet/resnet50/tico.py:121]: This comment does actually not apply to TiCo. Although we could try with the BYOL settings to check whether the accuracy increases.

File: README.md (hunk 1 above)
Comment (IgorSusmelj) [reply README.md:283]: 100 epochs vs 1000 epochs could be the reason for that
